sdk/keyvault/azure-security-keyvault/azure/security/keyvault/certificates/_models.py

```diff
@@ -0,0 +1,609 @@
+        # type: () -> list[str]
+        pass
+
+    @property
+    def subject_alternative_dns_names(self):
+        # type: () -> list[str]
+        pass
+
+    @property
+    def subject_alternative_upns(self):
+        # type: () -> list[str]
+        pass
+
+    @property
+    def validity_in_months(self):
+        # type: () -> int
+        pass
+
+    @property
+    def lifetime_actions(self):
+        pass
+
+    @property
+    def issuer_name(self):
+        # type: () -> str
+        pass
+
+    @property
+    def certificate_type(self):
+        # type: () -> str
+        pass
+
+    @property
+    def certificate_transparency(self):
+        # type: () -> bool
+        pass
+
+    @property
+    def created(self):
+        # type: () -> datetime
+        pass
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        pass
+
+    @property
+    def updated(self):
+        # type: () -> datetime
+        pass
+
+
+class Contact:
+    """The contact information for the vault certificates.
+    """
+
+    def __init__(self, email, name, phone):
+        # type: (str, str, str) -> None
+        self._email = email
+        self._name = name
+        self._phone = phone
+
+    @property
+    def email(self):
+        # type: () -> str
+        return self._email
+
+    @property
+    def name(self):
+        # type: () -> str
+        return self._name
+
+    @property
+    def phone(self):
+        # type: () -> str
+        return self._phone
+
+
+class IssuerBase(object):
+    def __init__(self, issuer_id, provider, attributes=None):
+        # type: (str, str, Optional[models.IssuerAttributes]) -> None
+        self._attributes = attributes
+        self._id = issuer_id
+        self._vault_id = _parse_vault_id(issuer_id)
+        self._provider = provider
+
+    @classmethod
+    def _from_issuer_item(cls, issuer_item):
+        # type: (models.CertificateIssuerItem) -> IssuerBase
+        """Construct a IssuerBase from an autorest-generated IssuerItem"""
+        return cls(issuer_id=issuer_item.id, provider=issuer_item.provider)
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        return self._attributes.enabled
+
+    @property
+    def created(self):
+        # type: () -> datetime
+        return self._attributes.created
+
+    @property
+    def updated(self):
+        # type: () -> datetime
+        return self._attributes.updated
+
+    @property
+    def id(self):
+        # type: () -> str
+        return self._id
+
+    @property
+    def name(self):
+        # type: () -> str
+        return self._vault_id.name
+
+    @property
+    def provider(self):
+        # type: () -> str
+        return self._provider
+
+    @property
+    def vault_url(self):
+        # type: () -> str
+        return self._vault_id.vault_url
+
+
+class Issuer(IssuerBase):
+    def __init__(
+        self,
+        attributes,
+        issuer_id,
+        provider,
+        account_id,
+        password,
```

Comment:
Are you certain none of these positional arguments could be optional? ("No, we can figure that out later" is a legitimate answer 😄)

Reply:
not certain -true.
Will update as I work on the implementation if the client.

Reply:
You have to look at the documentation. In many cases, the [openapi description](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.0/keyvault.json) of the API will indicate if the value is required or not. In some cases, the value is required under some circumstances (e.g. only when you create a resource), in which case it may not be explicitly specified as required in the openapi specification.

In this case, [it doesn't look like](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.0/keyvault.json#L4623) many of the attributes are actually required.